ip_country/src/ip_country.rs

```diff
@@ -273,4 +277,34 @@ Line 17: Invalid (AddrParseError(Ip)) IP address in CSV record: 'BOOGA'"#
 .to_string()
         );
     }
+
+    #[test]
+    fn write_error_test() {
+        let mut subject = BitQueue::new();
+        let output = &mut ByteArrayWriter::new();
+
+        subject.add_bits(0b11011, 5);
+        subject.add_bits(0b00111001110011100, 17);
+        subject.add_bits(0b1, 1);
+
```

Comment:
Lines from 285 to 289 seem doing more than what we need. 

if there is the NEED of pushing some bits into the queue first, then you can probably do the same by just a single call. For example by concatenating all the bit literals together and reevaluating the correct total amount of bits handled. In this case 20 is the total. 

if there is NO NEED of this filling stage, then just delete all three lines. 

Plus, at any rate, the blank lines of the ln.285 and 289, don't belong in there.